core/inst_handlers/b/RvzbaInsts.cpp

```diff
@@ -0,0 +1,168 @@
+        if constexpr (std::is_same_v<XLEN, RV64>)
+        {
+            inst_handlers.emplace(
+                "add.uw", atlas::Action::createAction<&RvzbaInsts::add_uw_handler, RvzbaInsts>(
+                            nullptr, "add.uw", ActionTags::EXECUTE_TAG));
+            inst_handlers.emplace(
+                "sh1add", atlas::Action::createAction<&RvzbaInsts::sh1add_handler<RV64>, RvzbaInsts>(
+                            nullptr, "sh1add", ActionTags::EXECUTE_TAG));
+            inst_handlers.emplace(
+                "sh1add.uw", atlas::Action::createAction<&RvzbaInsts::sh1add_uw_handler, RvzbaInsts>(
+                        nullptr, "sh1add.uw", ActionTags::EXECUTE_TAG));
+            inst_handlers.emplace(
+                "sh2add", atlas::Action::createAction<&RvzbaInsts::sh2add_handler<RV64>, RvzbaInsts>(
+                        nullptr, "sh2add", ActionTags::EXECUTE_TAG));
+            inst_handlers.emplace(
+                "sh2add.uw", atlas::Action::createAction<&RvzbaInsts::sh2add_uw_handler, RvzbaInsts>(
+                        nullptr, "sh2add.uw", ActionTags::EXECUTE_TAG));
+            inst_handlers.emplace(
+                "sh3add", atlas::Action::createAction<&RvzbaInsts::sh3add_handler<RV64>, RvzbaInsts>(
+                        nullptr, "sh3add", ActionTags::EXECUTE_TAG));
+            inst_handlers.emplace(
+                "sh3add.uw", atlas::Action::createAction<&RvzbaInsts::sh3add_uw_handler, RvzbaInsts>(
+                        nullptr, "sh3add.uw", ActionTags::EXECUTE_TAG));
+            inst_handlers.emplace(
+                "slli.uw", atlas::Action::createAction<&RvzbaInsts::slli_uw_handler, RvzbaInsts>(
+                        nullptr, "slli.uw", ActionTags::EXECUTE_TAG));
+        }
+        else if constexpr (std::is_same_v<XLEN, RV32>)
+        {
+        inst_handlers.emplace(
+            "sh1add", atlas::Action::createAction<&RvzbaInsts::sh1add_handler<RV32>, RvzbaInsts>(
+                nullptr, "sh1add", ActionTags::EXECUTE_TAG));
+        inst_handlers.emplace(
+            "sh2add", atlas::Action::createAction<&RvzbaInsts::sh2add_handler<RV32>, RvzbaInsts>(
+                nullptr, "sh2add", ActionTags::EXECUTE_TAG));
+        inst_handlers.emplace(
+            "sh3add", atlas::Action::createAction<&RvzbaInsts::sh3add_handler<RV32>, RvzbaInsts>(
+                nullptr, "sh3add", ActionTags::EXECUTE_TAG));
+        }
+    }
+
+    template void RvzbaInsts::getInstHandlers<RV32>(std::map<std::string, Action> &);
+    template void RvzbaInsts::getInstHandlers<RV64>(std::map<std::string, Action> &);
+
+    Action::ItrType RvzbaInsts::add_uw_handler(atlas::AtlasState* state, Action::ItrType action_it)
+    {
+        const AtlasInstPtr & inst = state->getCurrentInst();
+
+        // Ensures that rs1_val only receives the least-significant word of rs1
+        const uint64_t rs1_val = static_cast<uint64_t>(static_cast<uint32_t>(READ_INT_REG<uint64_t>(state, inst->getRs1())));

```

Comment:
Hmmm... wonder if this is better:
```suggestion
        const uint64_t rs1_val = READ_INT_REG<uint64_t>(state, inst->getRs1()) & 0xFFFFFFFFull;

```
I assume the `u` in `uw` is unsigned, so that should be ok.  We don't want to sign extend the u64.

Reply:
Yeah, that makes sense. There's quite a few other uw-esque instructions, so I'll go back and pair those down too